lib/rdf2marc/rdf2model/mappers/literal_or_remote_resolver.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Rdf2marc
+  module Rdf2model
+    module Mappers
+      # Mapping to Added Entry Fields model.
+      class LiteralOrRemoteResolver
+        # @param [RDF::Literal,RDF::URI] term
+        # @param [RDF::Graph] graph
+        # @param [Symbol] key_symbol one of (:personal_name, :corporate_name, :meeting_name, etc.)
+        # @param [Class] model the Ruby class to resolve for the model.
+        #                      (e.g. Rdf2marc::Models::General::PersonalName, etc.)
+        def self.resolve(term:, key_symbol:, graph:, model:)
+          return nil unless term
+          return { thesaurus: 'not_specified', key_symbol => term.value } if term.is_a?(RDF::Literal)
+
+          model = Resolver.resolve_model(term.value, model)
+
+          return model if model
+
+          result = { thesaurus: 'not_specified', authority_record_control_numbers: [term.value] }
+          label = graph.query(subject: term, predicate: RDF::RDFS.label).first&.object&.to_s
```

Comment:
For consistency:
```
label=item.work.query.path_first_literal([RDF::RDFS.label], subject_term: term)
```

Reply:
Are you suggesting that this method should take item as a parameter rather than graph?

Reply:
Yes, that would need to change to use this query.